adobe/brackets
# Issue #11261: keep the search bar open

_Reference:_
Issue: https://github.com/adobe/brackets/issues/11261 - "keep the search bar open"

Based on this pull request that is waiting on the user: https://github.com/adobe/brackets/pull/11692 ( @adrianhartanto0 ). However I ended up taking a different approach; upon testing I found that the original approach had hiccups. The changes are influenced by the preference idea by @zaggino and @abose in the aforementioned pull request.

_Changes:_
- Added in a menu option under View to allow find bar to stay open/automatically close
- Add a close icon to the search bar (for completeness)
- Fix escape key for in-editor searches
- Remember last search phrase if no selection for in-editor searches
- ~~Go to the next search item if the bar is open and the user clicks Find or hits ctrl+F~~

_Testing:_
- All tests pass for FindReplace-test.js after some changes to the code I wrote and one of the tests
- Manual testing was briefly performed as such:
  - In-Editor - Find the same item by resetting the cursor position: Pass
  - In-Editor - Find a custom item by setting the cursor position: Pass
  - In-Editor - Find the first item by setting the cursor position after the last: Pass
  - In-Editor - Cycle through the items by hitting ctrl+F while still open: Pass
  - In-Editor - Closes when the "Automatically close search" option is checked and the editor is focused: Pass
  - In-Editor - Moving to a different editor closes the search: Pass
  - In-Editor - Pre-selecting text and re-opening the search shows the newly selected text: Pass
  - In-Editor - Pre-selecting nothing and re-opening the search shows the last query: Pass
  - In-Editor - Selecting the caps and regexp options functions as expected: Pass
  - View --> Automatically close search - Preference is remembered on application restart: Pass
  - In-File: Search functions as expected, and the same tests for In-Editor, where applicable, also pass.
